### PR TITLE
Add binned BH mass - stellar mass observational data

### PR DESCRIPTION
**Pull request by FHusko**
This adds two data files and conversion scripts for a new BH mass - stellar mass plot. The plot is the same as the usual one, but the observational data (Graham & Sahu 2023 and Terrazas et al. 2017) are binned using fractions of morphological types as well as passive fractions. This then yields a precise median relation of BH mass and stellar mass that can be used for calibration.

### CODE REVIEW

**Comment by FHusko** [comment]
When I run the formatting script, I get the following error:
```
The typed_ast package is required but not installed.
You can upgrade to Python 3.8+ or install typed_ast with
`python3 -m pip install typed-ast`.
```
I have tried this with multiple versions of python, 3.8 and above. I even try installing typed-ast with different versions, the installation is done successfully, and yet I get the same error as above.

File: data/BlackHoleAccretionHistory/conversion/convertPouliasis2024_low.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+# Read the data
+raw = np.loadtxt(input_filename)
+z = unyt.unyt_array(raw[:, 0], "dimensionless")
+
+# Convert to scale factor
+a = 1 / (1 + z)
+
+BHARD_low = unyt.unyt_array(10**raw[:, 1], "Msun / yr / Mpc**3") 
+BHARD_high = unyt.unyt_array(10**raw[:, 2], "Msun / yr / Mpc**3")
```
Comment (robjmcgibbon) [inline data/BlackHoleAccretionHistory/conversion/convertPouliasis2024_low.py:32]: Similarly for this data, can you combine with the other script by doing a loop where you change the indices of the data you read in?

File: data/BlackHoleAccretionHistory/conversion/convertShen2020_high.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+# Read the data
+raw = np.loadtxt(input_filename)
+z = unyt.unyt_array(raw[:, 0], "dimensionless")
+
+# Convert to scale factor
+a = 1 / (1 + z)
+
+BLD = 10**unyt.unyt_array(raw[:, 4], "dimensionless") * unyt.erg / (unyt.s * unyt.Mpc**3)
+BLD_low = 10**unyt.unyt_array(raw[:, 5], "dimensionless") * unyt.erg / (unyt.s * unyt.Mpc**3)
+BLD_high = 10**unyt.unyt_array(raw[:, 6], "dimensionless") * unyt.erg / (unyt.s * unyt.Mpc**3)
```
Comment (robjmcgibbon) [inline data/BlackHoleAccretionHistory/conversion/convertShen2020_high.py:34]: Same for these two Shen scripts

File: data/GalaxyStellarMassBlackHoleMass/conversion/convertGraham2023_binned.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+output_filename = "Graham2023_binned.hdf5"
+output_directory = "../"
+
+if not os.path.exists(output_directory):
+    os.mkdir(output_directory)
+
+data = np.loadtxt("../raw/Graham2023_binned.txt")
+
+# Read in stellar masses
+M_star = (10 ** data[:, 0]) * unyt.Solar_Mass
```
Comment (robjmcgibbon) [inline data/GalaxyStellarMassBlackHoleMass/conversion/convertGraham2023_binned.py:24]: The convertGraham2023.py script has a stellar mass correction. Is that needed here too?

**Review by robjmcgibbon — CHANGES_REQUESTED** [review]
Couple of comments, mainly requesting you to combine the scripts with duplicated code. I was able to run the formatting script on cosma with the following modules `module load gnu_comp/11.1.0 openmpi/4.1.1 python/3.10.1`.

File: data/BlackHoleAccretionHistory/conversion/convertAnnana2020_low.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+# Convert to scale factor
+a = 1 / (1 + z)
+
+BLD = 10**unyt.unyt_array(raw[:, 1], "dimensionless") * unyt.erg / (unyt.s * unyt.Mpc**3)
+
+# Correct for cosmology
+BLD = BLD * (h_sim / h_obs) ** -2
+
+# Convert to BHARD
+radiative_efficiency = 0.34
```
[the remark this answers is not in the document]
Comment (FHusko) [reply data/BlackHoleAccretionHistory/conversion/convertAnnana2020_low.py:38]: Yes, that is the only difference, and indeed it is easy to create the two files through just one script. I've modified it now.

File: data/BlackHoleAccretionHistory/conversion/convertPouliasis2024_low.py (hunk 1 above)
Comment (FHusko) [reply data/BlackHoleAccretionHistory/conversion/convertPouliasis2024_low.py:32]: Same as above.

File: data/BlackHoleAccretionHistory/conversion/convertShen2020_high.py (hunk 2 above)
Comment (FHusko) [reply data/BlackHoleAccretionHistory/conversion/convertShen2020_high.py:34]: Done, same as above.

File: data/GalaxyStellarMassBlackHoleMass/conversion/convertGraham2023_binned.py (hunk 3 above)
Comment (FHusko) [reply data/GalaxyStellarMassBlackHoleMass/conversion/convertGraham2023_binned.py:24]: Thanks for spotting that, I hadn't noticed it! I didn't convert the stellar masses when binning, so I do it now in the script.

**Comment by robjmcgibbon** [comment]
Thanks! Could you please remove the _high and _low scripts using `git rm`, and then run the formatter over all scripts. If you're still having difficulty running it on cosma for some reason then I could do it.

**Comment by FHusko** [comment]
I've removed the duplicate scripts (sorry, thought I had already done it).

I can't seem to run the formatting script. With python 3.10.12 (I seem to need to use python > 3.6 in order to actually process all the observational data), I get the following warning when running the formatting script:
```
You can upgrade to Python 3.8+ or install typed_ast with
`python3 -m pip install typed-ast`.
```
This is despite using pyton >3.8 and having installed typed_ast.

**Comment by robjmcgibbon** [comment]
Could you please merge https://github.com/FHusko/velociraptor-comparison-data/pull/1

**Review by robjmcgibbon — APPROVED** [review]

**Pull request merged**